### PR TITLE
Release/2.1.0: merge changes and cleanup resources and Jenkins files

### LINKED ISSUE #48: Support Jenkins credentials import
For 2.1.0 current implementation environment is required to pass credentials, but we can try to improve it using a special environment definition to set credentials and then pass it as environment.

Jenkins allows to import credentials using withCredentials wrapper step. For the most generic purpose we already have support for tokens (string), tuple username and password, file, zip and certificate. This secrets can be imported from Jenkins Credentials store.

For that we can improve current release to allow the following:
```
config:
  project_repos:
    worsica-processing:
      repo: 'https://github.com/WORSICA/worsica-processing.git'
      branch: master
      dockerhub: worsica/worsica-processing
      dockertag: latest
  credentials:
    - type: string
      credentialsId: "myGithubCredentials"
      variable: "GITHUB_TOKEN"
    - type: string
      credentialsId: "myGithubCredentials2"
      variable: "GITHUB_TOKEN2"

sqa-criteria:
  qc-style:
    repos:
      worsica-processing:
        container: processing
        environment:
          - GITHUB_TOKEN: "${GITHUB_TOKEN}"
          - GITHUB_TOKEN2: "${GITHUB_TOKEN2}"
        tox:
          testenv:
            - stylecheck

environment:
  GIT_COMMITTER_NAME: Person1
  GIT_COMMITTER_EMAIL: person1@example.org
  LANG: C.UTF-8

timeout: 600
```

### LINKED ISSUE #62: Pipeline environment setup stage fails if project's name contains a white space
For instance:
docker-compose -f .sqa/docker-compose.yml --project-directory /var/lib/jenkins/workspace/EOSC Synergy/ up -d
No such command: Synergy/

The limitations on the project name should be documented as long as this issue is not fixed.

### PR DESCRIPTION
**Pull request by samuelbernardolip**
This pull request don't include the code for credentials finished. Is only the result of merging my previous work around release/2.1.0 with indigo-dc version.
I will create another two pull requests for:
- credentials ready version
- jenkins/R2.1.0 to test the library documentation

This PR overlaps draft PR #68 with additional corrections, so it should also be closed.
fix #62
resolve #48

### CODE REVIEW

**Comment by orviz** [comment]
@samuelbernardolip if you don't want this PR to be reviewed or merged you could turn it into a draft PR, so we know you are working on it but it is not ready to be reviewed

**Comment by samuelbernardolip** [comment]
> @samuelbernardolip if you don't want this PR to be reviewed or merged you could turn it into a draft PR, so we know you are working on it but it is not ready to be reviewed

@orviz My goal with this pull request is to ask your code review without testing with pipeline. I can't now change it to draft PR, at least I don't see the option to do the change...

File: src/eu/indigo/compose/DockerCompose.groovy [hunk 1]
```diff
@@ -66,10 +129,11 @@ class DockerCompose extends JenkinsDefinitions implements Serializable {
     * @param command Command with arguments to run inside container
     * @param args.composeFile Docker compose file to override the default docker-compose.yml
     * @param args.workdir Path to workdir directory for this command
+    * @param args.environment List with environment variables to be set in container runtime
```
Comment (orviz) [inline src/eu/indigo/compose/DockerCompose.groovy:132]: this param is not used

**Comment by samuelbernardolip** [comment]
Contrary to what I proposed before, I merged into this PR the changes for credentials.
This is already tested in jenkins with jpl-validator and is working.

File: src/eu/indigo/compose/DockerCompose.groovy (hunk 1 above)
Comment (samuelbernardolip) [reply src/eu/indigo/compose/DockerCompose.groovy]: Corrections applied as suggested by @orviz

File: src/eu/indigo/compose/DockerCompose.groovy [hunk 2]
```diff
@@ -18,19 +18,82 @@ class DockerCompose extends JenkinsDefinitions implements Serializable {
     */
     String _f = '-f'
     String _w = '--project-directory'
+    String _e = '-e'
```
[the remark this answers is not in the document]
Comment (samuelbernardolip) [reply src/eu/indigo/compose/DockerCompose.groovy]: Correction applied

**Pull request merged**